### DISCUSSION TITLE
Actual state of IMU implementation

### DISCUSSION THREAD

**Discussion by MartiBolet** [stella-cv/stella_vslam#211] [Q&A] [answered]
Good morning,
I'm actually trying to use an IMU with OpenVSLAM, as far as I saw, there's a branch with the implementation for EuRoC dataset example. I tried to use it in a real scenario with my own IMU, and seems it works, but when I try to load a pre-built map from zero fails to get the initial referenced inertial frame [`imu_preintegrator_from_inertial_ref_keyfrm_ == nullptr`]. I tried to solve this, but with my lack of knowledge about IMUs I don't know how to proceed.

So my question is, how is the state of the implementation of IMU integration? There's still much work to complete or when is expected to have full IMU integration in main branch? If I can help with anything I will try.

Thank you,
Martí Bolet Boixeda

**Comment by ymd-stella** [comment, marked as answer]
I have only tested SLAM mode. There is a lot of work to be done. The schedule is still undecided.

**Reply by Basavaraj-PN** [reply]
Hi @ymd-stella , any update on this?
